File: src/main/java/sc02/Q07.java
package sc02;

import java.io.BufferedReader;
import java.io.InputStreamReader;

/**
 * gooG
 * <p>
 * YES
 */

public class Q07 {
    public static void main(String[] args) {
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
            String line = br.readLine().toLowerCase();

            for (int i = 0; i < line.length() / 2; i++) {
                if (line.charAt(i) != line.charAt(line.length() - 1 - i)) {
                    System.out.println("NO");
                    return;
                }
            }
            System.out.println("YES");
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }
}

File: src/main/kotlin/sc02/Q07.kt
package jp.co.kopher.sc02

import java.io.BufferedReader
import java.io.IOException
import java.util.*

/**
 * gooG
 *
 * YES
 */

fun main() {
    try {
        BufferedReader(System.`in`.reader()).use { reader ->
            val line = reader.readLine().lowercase(Locale.getDefault())
            for (i in 0..line.length / 2) {
                if (line[i] != line[line.length - 1 - i]) {
                    println("NO")
                    return
                }
            }
            println("YES")
        }
    } catch (e: IOException) {
        e.printStackTrace()
    }
}